src/Models/NonhydrostaticModels/nonhydrostatic_model.jl

```diff
@@ -221,8 +221,9 @@ function inflate_grid_halo_size(grid, tendency_terms...)
 end
 
 # return the total advective velocities
-@inline total_velocities(m::NonhydrostaticModel) =
-    (u = SumOfArrays{2}(m.velocities.u, m.background_fields.velocities.u),
-     v = SumOfArrays{2}(m.velocities.v, m.background_fields.velocities.v),
-     w = SumOfArrays{2}(m.velocities.w, m.background_fields.velocities.w))
-
+@inline function total_velocities(m::NonhydrostaticModel)
+    velocities = (u = SumOfArrays{2}(m.velocities.u, m.background_fields.velocities.u),
+                  v = SumOfArrays{2}(m.velocities.v, m.background_fields.velocities.v),
+                  w = SumOfArrays{2}(m.velocities.w, m.background_fields.velocities.w))
+    return with_advective_forcing(m.forcing, velocities)
```

Comment:
I don't think this will work. `m.forcing` is a `NamedTuple` that contains different `Forcing` for every prognostic field (`forcing.u`, `forcing.v`, etc). When we call `with_advective_forcing`, we add the forcing per component, ie:

https://github.com/CliMA/Oceananigans.jl/blob/cc51ff0ee0a1326cebcbfa207ec0ed29ae768648/src/Models/NonhydrostaticModels/nonhydrostatic_tendency_kernel_functions.jl#L69

To be strictly correct we might have to consider all "effective" velocities experienced by each component, and take the minimum of all of those.

Since that's complicated (and unclear whether we even want to spare the expense for that kind of calculation), I'd leave it for the future. Typically, "additional" forcing-associated velocities (like settling velocities) are known in advance and could be incorporated into the user determination of the cfl.

Reply:
I guess it's also unclear what the "total velocities" mean when different tracers can have different effective velocities. Maybe it's better to leave the forcing-related velocity out of total velocities for clarity...

Reply:
That sounds good, I will remove the second commit related to `AdvectiveForcing`.